### PR TITLE
Build comment section

### PR DESCRIPTION
**Pull request by chen678**
Front end part: 
- Build comment section UI, from top to bottom:
	-	Welcome message
	-	Comment history or "leave a message" if the list is empty
	-	Box for message input
	-	Submit button.
- After submission, redirect to the comment anchor link (for UX consistency).

Back end part:
- GET and POST methods.
- Comment history objects. 

Todo: record message creating time and display it in the front end.

Preview/Screenshots:
First time loading:
Screen Shot 2020-02-27 at 10 44 01 PM
After you leave a message:
Screen Shot 2020-02-27 at 10 44 12 PM

### CODE REVIEW

**Comment by chen678** [comment]
Updated UI after creating datastore:
Screen Shot 2020-02-28 at 1 41 36 AM

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.google.sps.data;
+
+import java.util.Date;
+
+/** A comment entry. */
```
Comment (kmontgomery) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: Update this comment.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -26,28 +27,28 @@
...
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String comment =  request.getParameter("user-comment");
+        if (comment.length() == 0) {
+            response.setContentType("text/html");
+            response.getWriter().println("Your comment is empty!");
+            return;
+        }
```
Comment (kmontgomery) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: Really nice job handling an edge case. When we start writing unit tests, this is a good example of something to test.

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -11,14 +11,15 @@
     <script src="js/script.js"></script>
   </head>
   
-  <body onload = "fetchJson()">
```
Comment (kmontgomery) [inline portfolio/src/main/webapp/index.html:14]: nice job changing the title of this method to reflect the updated behavior.

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -50,7 +51,7 @@ <h1>
             </h1>
             <p>
                 Some of the projects I have worked on: <br />
-                Android app: story sticker; orchardwatch (app for local apple farm) <br />
+                Android app: story creation page in Facebook story; orchardwatch (app for local apple farm) <br />
```
Comment (kmontgomery) [inline portfolio/src/main/webapp/index.html:54]: small nit: this change shouldn't technically be in this PR since it's unrelated. Fine to leave for now though!

File: portfolio/src/main/webapp/js/script.js [hunk 5]
```diff
@@ -24,10 +24,44 @@ function responsiveTopNav() {
...
-        console.log(array);
-        document.getElementById('fetch-result').innerText = array;
+/** Fetches the list of all user comments and builds the UI. */
+function getComments(){
+    fetch('/list-comments').then(response => response.json()).then((comments) => {
+        // Build the list of history comments.
+        const historyOfComments = document.getElementById('history');
+        console.log(comments);
+        if(comments.length == 0){
+            historyOfComments.appendChild(createListElement("OH! Be the first one to leave a message!"));
```
Comment (kmontgomery) [inline portfolio/src/main/webapp/js/script.js:33]: Nice touch :)

File: portfolio/src/main/webapp/js/script.js [hunk 6]
```diff
@@ -24,10 +24,44 @@ function responsiveTopNav() {
...
+        } else{
+            comments.forEach((entry) => {
+                console.log(entry);
```
Comment (kmontgomery) [inline portfolio/src/main/webapp/js/script.js]: It usually not best practice to log user input as this could be considered personal information. Of course, its fine for this case, but just something to be aware of!

**Review by kmontgomery — APPROVED** [review]
A few small nits, but looks great!

File: portfolio/src/main/webapp/js/script.js (hunk 6 above)
Comment (chen678) [reply portfolio/src/main/webapp/js/script.js]: agreed. the log was for debugging purposes. They are deleted in the new commit.

**Comment by chen678** [comment]
addressed comments.
